### PR TITLE
Add-ons built with Makefile, build.sh added (GCI 2016)

### PR DESCRIPTION
**Pull request by CodeforEvolution**
This commit will make add-ons now use the Makefile system and as a bonus have added a ./build.sh to make it easy for someone else to change the haikuports recipe to build the add-ons. I have left other notes back at the Code-In task. Hope this pleases! :)

### CODE REVIEW

File: addons/UtilityClasses/ImageProcessingLibrary.cpp [hunk 1]
```diff
@@ -166,7 +166,7 @@ status_t ImageProcessingLibrary::gaussian_blur(BBitmap *bitmap,float radius, int
 	int32 d_bpr = intermediate->BytesPerRow()/4;
 
 	// Start the threads.
-	thread_id blur_thread_array[B_MAX_CPU_COUNT];
+	thread_id blur_thread_array[8];
```
Comment (waddlesplash) [inline addons/UtilityClasses/ImageProcessingLibrary.cpp]: Might want to add a TODO about deciding how many threads to use based on CPU count.

File: artpaint/paintwindow/PaintWindow.cpp [hunk 2]
```diff
@@ -48,6 +48,7 @@
 #include <Button.h>
 #include <Clipboard.h>
 #include <Entry.h>
+#include <LayoutBuilder.h>
```
Comment (waddlesplash) [inline artpaint/paintwindow/PaintWindow.cpp]: You don't seem to actually use this?

**Comment by scottmc** [comment]
For issues you'd turn up, please open a new issue here.  The notes on the GCI task will be lost once the contest closed as Google removes the data left there when the contest is over.

**Comment by CodeforEvolution** [comment]
I have added the fixes wanted, also I will open up issues for the broken addons here and an issue for updating the recipe at haikuports when I can. Hope this helps!

**Comment by CodeforEvolution** [comment]
This should close issue #43 if this is considered done.

File: Makefile [hunk 3]
```diff
@@ -141,12 +141,12 @@ WARNINGS =
...
 #	so that stack crawls in the debugger are meaningful
 #	if TRUE symbols will be created
-SYMBOLS = 
+SYMBOLS = TRUE
 
 #	specify debug settings
 #	if TRUE will allow application to be run from a source-level
 #	debugger.  Note that this will disable all optimzation.
-DEBUGGER = 
+DEBUGGER = TRUE
```
Comment (diversys) [inline Makefile]: Perhaps leave it disabled by default?

**Comment by CodeforEvolution** [comment]
Alright! Did that! Also made the .gitignore more flexible to people not using the build.sh script

**Comment by CodeforEvolution** [comment]
Is this considered ready now to be merged? Anything else I can do?

**Comment by CodeforEvolution** [comment]
Thank you!

**Pull request merged**